dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNLineInterfaceIIDM.cpp

```diff
@@ -0,0 +1,274 @@
+
+#include "DYNTrace.h"
+
+using boost::shared_ptr;
+using std::string;
+using std::vector;
+
+namespace DYN {
+
+LineInterfaceIIDM::LineInterfaceIIDM(powsybl::iidm::Line& line) : lineIIDM_(line),
+                                                                  initialConnected1_(boost::none),
+                                                                  initialConnected2_(boost::none) {
+  setType(ComponentInterface::LINE);
+  stateVariables_.resize(5);
+  stateVariables_[VAR_P1] = StateVariable("p1", StateVariable::DOUBLE);     // P1
+  stateVariables_[VAR_P2] = StateVariable("p2", StateVariable::DOUBLE);     // P2
+  stateVariables_[VAR_Q1] = StateVariable("q1", StateVariable::DOUBLE);     // Q1
+  stateVariables_[VAR_Q2] = StateVariable("q2", StateVariable::DOUBLE);     // Q2
+  stateVariables_[VAR_STATE] = StateVariable("state", StateVariable::INT);  // connectionState
+}
+
+LineInterfaceIIDM::~LineInterfaceIIDM() {
+}
+
+void
+LineInterfaceIIDM::setBusInterface1(const shared_ptr<BusInterface>& busInterface) {
+  busInterface1_ = busInterface;
+}
+
+void
+LineInterfaceIIDM::setVoltageLevelInterface1(const shared_ptr<VoltageLevelInterface>& voltageLevelInterface) {
+  voltageLevelInterface1_ = voltageLevelInterface;
+}
+
+void
+LineInterfaceIIDM::setBusInterface2(const shared_ptr<BusInterface>& busInterface) {
+  busInterface2_ = busInterface;
+}
+
+void
+LineInterfaceIIDM::setVoltageLevelInterface2(const shared_ptr<VoltageLevelInterface>& voltageLevelInterface) {
+  voltageLevelInterface2_ = voltageLevelInterface;
+}
+
+shared_ptr<BusInterface>
+LineInterfaceIIDM::getBusInterface1() const {
+  return busInterface1_;
+}
+
+shared_ptr<BusInterface>
+LineInterfaceIIDM::getBusInterface2() const {
+  return busInterface2_;
+}
+
+double
+LineInterfaceIIDM::getVNom1() const {
+  if (lineIIDM_.getTerminal1().isConnected()) {
+    return lineIIDM_.getTerminal1().getVoltageLevel().getNominalVoltage();
+  }
+  return 0;
+}
+
+double
+LineInterfaceIIDM::getVNom2() const {
+  if (lineIIDM_.getTerminal2().isConnected()) {
+    return lineIIDM_.getTerminal2().getVoltageLevel().getNominalVoltage();
+  }
+  return 0;
+}
+
+double
+LineInterfaceIIDM::getR() const {
+  return lineIIDM_.getR();
+}
+
+double
+LineInterfaceIIDM::getX() const {
+  if (doubleIsZero(lineIIDM_.getX()) && doubleIsZero(lineIIDM_.getR())) {
+    Trace::warn() << DYNLog(PossibleDivisionByZero, lineIIDM_.getId()) << Trace::endline;
+    return 0.01;  // default parameter
+  }
+  return lineIIDM_.getX();
+}
+
+double
+LineInterfaceIIDM::getB1() const {
+  return lineIIDM_.getB1();
+}
+
+double
+LineInterfaceIIDM::getB2() const {
+  return lineIIDM_.getB2();
+}
+
+double
+LineInterfaceIIDM::getG1() const {
+  return lineIIDM_.getG1();
+}
+
+double
+LineInterfaceIIDM::getG2() const {
+  return lineIIDM_.getG2();
+}
+
+double
+LineInterfaceIIDM::getP1() {
+  double P = 0;
+  if (getInitialConnected1()) {
+    if (!std::isnan(lineIIDM_.getTerminal1().getP())) {
```

Comment:
Inverted: should be 'std::isnan' and not '! std::isnan'

Reply:
(same for others similar methods)

Reply:
SUre, I did not start testing that part !